com.reprezen.swagedit/src/com/reprezen/swagedit/assist/JsonReferenceProposalProvider.java

```diff
@@ -0,0 +1,288 @@
+				return PATH_PARAMETER;
+			} else if (path.matches(RESPONSE_REGEX)) {
+				return PATH_RESPONSE;
+			} else if (path.matches(PATH_ITEM_REGEX)) {
+				return PATH_ITEM;
+			}
+
+			return UNKNOWN;
+		}
+	}
+
+	protected Collection<JsonNode> collectPathItems(JsonNode document, IPath path) {
+		Collection<JsonNode> results = Lists.newArrayList();
+
+		JsonNode parameters = document.get("paths");
+		if (parameters == null) {
+			return results;
+		}
+
+		for (Iterator<String> it = parameters.fieldNames(); it.hasNext();) {
+			String key = it.next();			
+			String value =  (path != null ? path.toString() : "") 
+					+ "#/paths/" + key.replaceAll("/", "~1");
+
+			results.add(mapper.createObjectNode()
+					.put("value", "\"" + value + "\"")
+					.put("label", key)
+					.put("type", value) );
+		}
+
+		return results;
+	}
+
+	protected Collection<JsonNode> collectParameters(JsonNode document, IPath path) {
+		Collection<JsonNode> results = Lists.newArrayList();
+
+		JsonNode parameters = document.get("parameters");
+		if (parameters == null) {
+			return results;
+		}
+
+		for (Iterator<String> it = parameters.fieldNames(); it.hasNext();) {
+			String key = it.next();
+			String value =  (path != null ? path.toString() : "") + "#/parameters/" + key;
+
+			results.add(mapper.createObjectNode()
+					.put("value", "\"" + value + "\"")
+					.put("label", key)
+					.put("type", value) );
+		}
+
+		return results;
+	}
+
+	protected Collection<JsonNode> collectResponses(JsonNode document, IPath path) {
+		Collection<JsonNode> results = Lists.newArrayList();
+
+		JsonNode parameters = document.get("responses");
+		if (parameters == null) {
+			return results;
+		}
+
+		for (Iterator<String> it = parameters.fieldNames(); it.hasNext();) {
+			String key = it.next();
+			String value =  (path != null ? path.toString() : "") + "#/responses/" + key;
+
+			results.add(mapper.createObjectNode()
+					.put("value", "\"" + value + "\"")
+					.put("label", key)
+					.put("type", value) );
+		}
+
+		return results;
+	}
+
+	protected Collection<JsonNode> collectDefinitions(JsonNode document, IPath path) {
+		Collection<JsonNode> results = Lists.newArrayList();
+
+		if (document.has("definitions")) {
+			JsonNode definitions = document.get("definitions");
+
+			for (Iterator<String> it = definitions.fieldNames(); it.hasNext();) {
+				String key = it.next();
+				String value =  (path != null ? path.toString() : "") + "#/definitions/" + key;
+
+				results.add(mapper.createObjectNode()
+						.put("value", "\"" + value + "\"")
+						.put("label", key)
+						.put("type", value) );
+			}
+		}
+
+		return results;
+	}
+
+	private static class FileVisitor implements IResourceProxyVisitor {
+
+		private final List<IFile> files = new ArrayList<>();
+
+		@Override
+		public boolean visit(IResourceProxy proxy) throws CoreException {
+			if (proxy.getType() == IResource.FILE && 
+					(proxy.getName().endsWith("yaml") || proxy.getName().endsWith("yml"))) {
```

Comment:
Can we reuse the file extensions (`yaml` and `yml`) from somewhere else? Or at least declare the constants in a more visible place?